pkg/interop/convert/convert.go

```diff
@@ -20,3 +20,247 @@ func ToString(v any) string {
 func ToBool(v any) bool {
 	return v.(bool)
 }
+
+// UintToBEBytes converts an uint64 to a slice of a specified length
+// in big-endian format.
+func UintToBEBytes(m uint64, width uint64) []byte {
+	// without creating a new variable, compilation fails with panic, check (#3993)
+	var n = m
+	var res = make([]byte, width)
+	for i := int64(width - 1); i >= 0; i-- {
+		res[i] = byte(n & 0xff)
+		n = n >> 8
+	}
+	return res
+}
+
+// Uint64ToBEBytes converts an uint64 to a fixed-length slice in big-endian format.
+func Uint64ToBEBytes(n uint64) []byte {
+	return UintToBEBytes(n, 8)
+}
+
+// Uint32ToBEBytes converts an uint32 to a fixed-length slice in big-endian format.
+func Uint32ToBEBytes(n uint32) []byte {
+	return UintToBEBytes(uint64(n), 4)
+}
+
+// Uint16ToBEBytes converts an uint16 to a fixed-length slice in big-endian format.
+func Uint16ToBEBytes(n uint16) []byte {
+	return UintToBEBytes(uint64(n), 2)
+}
+
+// Uint8ToBEBytes converts an uint8 to a fixed-length slice in big-endian format.
+func Uint8ToBEBytes(n uint8) []byte {
+	return UintToBEBytes(uint64(n), 1)
+}
```

Comment:
We likely don't need this one, it's trivial. @roman-khimov?

Reply:
It has no BE/LE concept (which should be reflect d in naming), but `[]byte{0}` vs `[]byte{}` is still important.